aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -139,20 +139,33 @@ class DispatchTable final {
    * Perform a dynamic dispatch on this table and find the kernel to call
    * for the given arguments.
    *
-   * @param args Arguments to invoke the function with
+   * @param stack Stack with arguments to invoke the kernel function with
    * @return Kernel function pointing to the right kernel for the given arguments.
    */
    const KernelFunction& lookup(const Stack* stack) const {
-     return lookup_([=] () -> c10::optional<TensorTypeId> {
+     return lookup_([&] () -> c10::optional<TensorTypeId> {
        if (!dispatch_strategy_.is_valid_) {
          return c10::nullopt;
        }
-       return dispatch_strategy_.get_dispatch_key(stack, operator_name_);
+       return dispatch_strategy_.get_dispatch_key(stack);
      });
    }
 
-   const KernelFunction& lookup(TensorTypeId dispatchKey) const {
-     return lookup_([=] () -> c10::optional<TensorTypeId> { return dispatchKey;});
+   /**
+    * Perform a dynamic dispatch on this table and find the kernel to call
+    * for the given arguments.
+    *
+    * @param args Arguments to invoke the kernel function with
+    * @return Kernel function pointing to the right kernel for the given arguments.
+    */
+   template<class... Args>
+   const KernelFunction& lookup(const Args&... args) const {
+     return lookup_([&] () -> c10::optional<TensorTypeId> {
+       if (!dispatch_strategy_.is_valid_) {
```

Comment:
Still not sure why we need to indirect through a dispatch strategy

Reply:
I'm currently working on another simplification diff that will probably address that